packages/components/src/suggestions/item.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import type { ForwardRefRenderFunction, FocusEvent, MouseEvent } from 'react';
+import classNames from 'classnames';
+import { escapeRegExp } from 'lodash';
+
+function escapeRegExpWithSpace( str: string ) {
+	return escapeRegExp( str ).replace( /\s/g, '\\s' );
+}
+
+const createTextWithHighlight = ( text: string, query: string ) => {
+	const re = new RegExp( '(' + escapeRegExpWithSpace( query ) + ')', 'gi' );
+	const parts = text.split( re );
+
+	// Replaces char code 160 (&nbsp;) with 32 (space)
+	const match = query.toLowerCase().replace( /\s/g, ' ' );
+
+	return parts.map( ( part, i ) => {
+		const key = text + i;
+		const lowercasePart = part.toLowerCase();
+		const spanClass = classNames( 'suggestions__label', {
+			'is-emphasized': lowercasePart === match,
+		} );
+
+		return (
+			<span key={ key } className={ spanClass }>
+				{ part }
+			</span>
+		);
+	} );
+};
+
+type Props = {
+	label: string;
+	hasHighlight: boolean;
+	query: string;
+	onMount: () => void;
+	onMouseDown: () => void;
+	onMouseOver: () => void;
+};
+
+const InnerItem: ForwardRefRenderFunction< HTMLButtonElement, Props > = (
+	{ label, hasHighlight, query, onMount, onMouseDown, onMouseOver },
+	forwardedRef
+) => {
+	React.useEffect( () => {
+		onMount();
+	}, [ onMount ] );
```

Comment:
Is this functionally different since the original component wouldn't re-call `onMount` if `onMount` changed? Not sure why it _would_ change though

Reply:
> Not sure why it would change though

That was my thought exactly when writing this... I guess we should make it fully 1:1 though and remove it from the dependencies.